File: services/photo_subscription.py
"""Photo Analysis Subscription Service"""
from ravendb import DocumentStore
from ravendb.documents.subscriptions.document_subscriptions import SubscriptionCreationOptions


class PhotoSubscription:
    """Subscription for photo analysis notifications"""
    
    SUBSCRIPTION_NAME = "After Photos Analysis"
    
    @classmethod
    def create(cls, store: DocumentStore):
        """Create the photo analysis subscription if it doesn't exist"""
        try:
            # Try to get existing subscription
            existing = store.subscriptions.get_subscription_state(cls.SUBSCRIPTION_NAME)
            if existing:
                print(f"Subscription '{cls.SUBSCRIPTION_NAME}' already exists")
                return
        except Exception:
            # Subscription doesn't exist, create it
            pass
        
        store.subscriptions.create_for_options(
            SubscriptionCreationOptions(
                name=cls.SUBSCRIPTION_NAME,
                query='from "Photos" where Description != null'
            )
        )

File: Services/PhotoSubscription.cs
using Raven.Client.Documents;
using Raven.Client.Documents.Subscriptions;
using Raven.Client.Exceptions.Documents.Subscriptions;

namespace PropertySphere.Services;

public static class PhotoSubscription
{
    private const string SubscriptionName = "After Photos Analysis";

    public static void Create(IDocumentStore store)
    {
        try
        {
            store.Subscriptions.GetSubscriptionState(SubscriptionName);
            Console.WriteLine($"Subscription '{SubscriptionName}' already exists");
            return;
        }
        catch (SubscriptionDoesNotExistException)
        {
            // Subscription doesn't exist, create it
        }

        store.Subscriptions.Create(new SubscriptionCreationOptions
        {
            Name = SubscriptionName,
            Query = """
                from "Photos" 
                where Description != null
                """
        });
    }
}
